### PR TITLE
[refactor] #1210: Split event stream

### PR DESCRIPTION
**Pull request by mversic**
### Description of the Change

Split `EventSocketMessage` into separate enums for transmitting / receiving

### Benefits

Less chance a wrong message will be sent over the socket

### CODE REVIEW

File: data_model/src/events.rs [hunk 1]
```diff
@@ -1,63 +1,85 @@
...
-    /// Request sent by client to subscribe to events.
-    SubscriptionRequest(SubscriptionRequest),
-    /// Answer sent by peer.
-    /// The message means that all event connection is initialized and will be supplying
-    /// events starting from the next one.
+/// Message sent by the stream producer
+#[version_with_scale(n = 1, versioned = "VersionedEventPublisherMessage")]
+#[derive(Debug, Clone, Decode, Encode, FromVariant, IntoSchema)]
+pub enum EventPublisherMessage {
+    /// Answer sent by the peer.
```
Comment (appetrosyan) [inline data_model/src/events.rs:37]:
```suggestion
    /// Reply sent by the peer.
```

**Comment by appetrosyan** [comment]
This shares a lot with #1724.

**Comment by mversic** [comment]
> This shares a lot with #1724.

I know, the other one depends on this. It will be sorted out when this is merged and I'll remove `draft` label from the other

File: data_model/src/events.rs (hunk 1 above)
Comment (mversic) [reply data_model/src/events.rs:37]: yes

**Review by s8sato — APPROVED** [review]

**Review by s8sato — APPROVED** [review]

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**